### PR TITLE
User Manual update for HWLoops and exception handlers

### PR DESCRIPTION
**Pull request by pascalgouedo**

### CODE REVIEW

**Review by davideschiavone — APPROVED** [review]

File: docs/source/corev_hw_loop.rst [hunk 1]
```diff
@@ -138,3 +138,46 @@ it is executed 10x10 times. Whereas the outermost loop, from startO to (endO - 4
...
+
+Application and ebreak/ecall exception handlers
+-----------------------------------------------
+
+When an ebreak or an ecall instruction is used in an application, special care should be given for those instruction handlers in case they are placed as the last instruction of an HWLoop.
+Those handlers should manage MEPC and lpcountX CSRs updates because an hw loop early-exit could happen if not done.
+
+At the end of the handlers after restoring the context/CSRs, a piece of smart code should be added (by order of piority):
+
+1. if MEPC = "lpend0 - 4" and lpcount0 >= 2 then MPEC should be set to lpstart0; if MEPC = "lpend0 - 4" and lpcount0 >= 1 then it should be decremented by 1.
```
Comment (davideschiavone) [inline docs/source/corev_hw_loop.rst:154]: please specify priority

File: docs/source/corev_hw_loop.rst [hunk 2]
```diff
@@ -138,3 +138,46 @@ it is executed 10x10 times. Whereas the outermost loop, from startO to (endO - 4
...
+2. if MEPC = "lpend1 - 4" and lpcount1 >= 2 then MPEC should be set to lpstart1; if MEPC = "lpend1 - 4" and lpcount1 >= 1 then it should be decremented by 1.
+3. if (lpstart0 <= MEPC < lpend0 - 4) or (lpstart1 <= MEPC < lpend1 - 4) then MPEC should be incremented by 4.
+4. if instruction at MEPC location is either ecall or ebreak then MPEC should be incremented by 4.
+5. if instruction at MEPC location location is c.ebreak then MPEC should be incremented by 2.
```
Comment (davideschiavone) [inline docs/source/corev_hw_loop.rst:158]: please specify priority and IF ELSE between 1 to 5

File: docs/source/corev_hw_loop.rst [hunk 3]
```diff
@@ -138,3 +138,46 @@ it is executed 10x10 times. Whereas the outermost loop, from startO to (endO - 4
...
+
+The 2 last cases are the standard ones when ebreak/ecall are not inside an HWLopp.
+
+Interrupt handlers
+------------------
+
+When an interrupt is happening on the last HWLoop instruction, its execution is cancelled, its address is saved in MEPC and its execution will be resumed when returning from interrupt handler.
```
Comment (davideschiavone) [inline docs/source/corev_hw_loop.rst:165]: Why should state also here for clarity that these registers (HWLoops) need to be saved and restored in/from the stack

**Review by davideschiavone — CHANGES_REQUESTED** [review]

File: docs/source/corev_hw_loop.rst (hunk 3 above)
Comment (pascalgouedo) [reply docs/source/corev_hw_loop.rst:164]: Done.

**Review by davideschiavone — APPROVED** [review]

**Pull request merged**